### PR TITLE
Emily/baseline model organization

### PR DESCRIPTION
**Pull request by xehu**
This PR adds two functional modifications to the featurization / model pipeline:

1. **Add the ability to report only TRAIN + VALIDATION (no test set).** Previously, we report 3 splits: train, validation, and test. This PR adds the ability to select only a train + validation split, by setting the train_size to None.
2. **Add the ability to analyze data from only the first x% of a conversation.** Sometimes, we wish to predict a conversation's outcome from only the first part of a conversation --- partly because of the path-dependency of conversation (the last part is dependent on the first part!) and partly because the ending can "spoil" the prediction by explicitly stating the result (e.g., "wow, guys, look at our high score!"; "hey, I guess we are all in agreement in this jury"). In the Feature Builder stage, we are now able to specify a truncation percentage, such that we analyze only the first x% of the conversation.

It also adds a few organizational changes to the model:
1. It moves all things related to the model to the `model/` folder;
2. The baselines are split across two specific iPython notebooks -- `model_per_task_baselines` (modeling each task separately); and `model_baseline_across_all_tasks` (modeling all tasks together);
3. We specify and run all baselines.

### CODE REVIEW

File: feature_engine/feature_builder.py [hunk 1]
```diff
@@ -27,7 +27,8 @@ def __init__(
             self, 
             input_file_path: str, 
             output_file_path_chat_level: str, 
-            output_file_path_conv_level: str
+            output_file_path_conv_level: str,
+            analyze_first_pct: float=1.0
```
Comment (xehu) [inline feature_engine/feature_builder.py:31]: Note that, by default, `analyze_first_pct` is set to 1, so that we have backwards compatibility; running the previous models as-is leads to the same output.

File: feature_engine/model/config.json [hunk 2]
```diff
@@ -53,7 +53,8 @@
         "csop": "Room assignment task",
         "csopII": "Room assignment task",
         "dat": "Divergent Association Task",
-        "becker": "Estimating Factual Quantities"
+        "becker": "Estimating Factual Quantities",
+        "gurcay": "Estimating Factual Quantities"
```
Comment (xehu) [inline feature_engine/model/config.json:57]: necessary because gurcay is also an estimation task!

File: feature_engine/model/model_builder.py [hunk 3]
```diff
@@ -366,18 +366,31 @@ def create_holdout_sets(self, val_size: float=0.1, test_size: float=None) -> Non
         """
         # Calls the `define_dataset_for_model()` function to get the `X`s and the `y`s
         self.X, self.y = self.define_dataset_for_model()
+        
         # If we need to split out a test set
-        if test_size:
+        if self.test_dataset_names == None:
```
Comment (xehu) [inline feature_engine/model/model_builder.py:371]: changing this if condition to mean, "if we need to split out a test set, because there is no external test dataset"

File: feature_engine/model/model_builder.py [hunk 4]
```diff
@@ -366,18 +366,31 @@ def create_holdout_sets(self, val_size: float=0.1, test_size: float=None) -> Non
...
             # Spit train-val and test set in a `(1-test_size)-test_size` split
-            X_train_val, X_test, y_train_val, y_test = train_test_split(self.X, self.y, random_state=42, test_size=test_size)
+            X_train_val = self.X
+            y_train_val = self.y
+            
+            # Split test set only if specified
+            if test_size:
```
Comment (xehu) [inline feature_engine/model/model_builder.py:377]: testing for the special case here, where we actually need the separate test dataset
